service/handlers.go

```diff
@@ -0,0 +1,149 @@
+package service
+
+import (
+	"encoding/json"
+	"fmt"
+	log "github.com/Sirupsen/logrus"
+	"github.com/rancher/go-rancher/v2"
+	"io/ioutil"
+	"net/http"
+	"time"
+)
+
+type WebhookRequest map[string]interface{}
+
+func Construct(w http.ResponseWriter, r *http.Request) {
+	var webhookRequestData WebhookRequest
+	log.Infof("Construct Payload")
+	bytes, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		log.Errorf("Construct failed with error: %v", err)
+		return
+	}
+	json.Unmarshal(bytes, &webhookRequestData)
+	_, err = ConstructPayload(webhookRequestData)
+}
+
+// ConstructPayload accepts and sends request to driver
+func ConstructPayload(parameters map[string]interface{}) (map[string]interface{}, error) {
```

Comment:
See my comments below to refactor this into a a generic framework for calling drivers.
When you're done, there should be no references to specific drivers like service in this function.